### PR TITLE
[MM-31398] upgrade error level to critical in case of panic or os.Exit

### PR DESCRIPTION
**Pull request by isacikgoz**
#### Summary
We should use `mlog.Critical` if we are going to call `panic` or `os.Exit(>0)`

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-31398

#### Release Note

```release-note
NONE
```

### CODE REVIEW

File: api4/apitestlib.go [hunk 1]
```diff
@@ -609,7 +609,7 @@ func (th *TestHelper) CreateDmChannel(user *model.User) *model.Channel {
 	var err *model.AppError
 	var channel *model.Channel
 	if channel, err = th.App.GetOrCreateDirectChannel(th.BasicUser.Id, user.Id); err != nil {
-		mlog.Error(err.Error())
+		mlog.Critical(err.Error())
 
 		time.Sleep(time.Second)
```
Comment (agnivade) [inline api4/apitestlib.go:614]: These are all test code. I would just say to remove the `mlog.Error` and `time.Sleep` and to just let it panic.

File: testlib/helper.go [hunk 2]
```diff
@@ -169,7 +169,7 @@ func (h *MainHelper) PreloadMigrations() {
 	handle := h.SQLStore.GetMaster()
 	_, err = handle.Exec(string(buf))
 	if err != nil {
-		mlog.Error("Error preloading migrations. Check if you have &multiStatements=true in your DSN if you are using MySQL. Or perhaps the schema changed? If yes, then update the warmup files accordingly.")
+		mlog.Critical("Error preloading migrations. Check if you have &multiStatements=true in your DSN if you are using MySQL. Or perhaps the schema changed? If yes, then update the warmup files accordingly.")
```
Comment (agnivade) [inline testlib/helper.go:172]: We can also do a errors.Wrap(msg, err) and feed that to panic. I wrote this code. I just wanted this custom error message to appear along with the original error. We don't really need two separate logs. Single panic with the combined error message is sufficient.

File: api4/apitestlib.go (hunk 1 above)
Comment (isacikgoz) [reply api4/apitestlib.go:614]: Fair enough 👍

**Review by streamer45 — APPROVED** [review]
Nice :+1: Nothing to add to @agnivade's comments.

File: testlib/helper.go [hunk 3]
```diff
@@ -169,8 +170,7 @@ func (h *MainHelper) PreloadMigrations() {
 	handle := h.SQLStore.GetMaster()
 	_, err = handle.Exec(string(buf))
 	if err != nil {
-		mlog.Error("Error preloading migrations. Check if you have &multiStatements=true in your DSN if you are using MySQL. Or perhaps the schema changed? If yes, then update the warmup files accordingly.")
-		panic(err)
+		panic(errors.Wrap(err, "error preloading migrations. check if you have &multiStatements=true in your DSN if you are using MySQL. or perhaps the schema changed? if yes, then update the warmup files accordingly"))
```
Comment (agnivade) [inline testlib/helper.go:173]: This is a bit tricky. I realize error strings should not be capitalized. But we have 4 sentences here. It looks really odd if the next letter is small after a period. Let's keep using capitals as an exception here.

**Review by agnivade — APPROVED** [review]
Just one small comment.

File: testlib/helper.go (hunk 3 above)
Comment (isacikgoz) [reply testlib/helper.go:173]: I was caught in the middle for this 😅

**Pull request merged**